### PR TITLE
MULE-9934: Extract HTTP response creation logic from response builder…

### PR DESCRIPTION
**Pull request by afelisatti**
… component

### CODE REVIEW

File: tests/http/src/test/resources/http-request-query-params-config.xml [hunk 1]
```diff
@@ -47,8 +47,8 @@
         <httpn:request config-ref="config" path="testPath"  key="ANY">
             <httpn:request-builder>
                 <!-- Not currently supported. -->
-                <!--<httpn:requester-request-builder queryParams="#[{'testName1':null, 'testName2': [null]}]"/>-->
-                <httpn:requester-request-builder queryParams="#[{'testName1':null, 'testName2': null}]"/>
+                <!--<httpn:simple-request-builder queryParams="#[{'testName1':null, 'testName2': [null]}]"/>-->
```
Comment (pablokraan) [inline tests/http/src/test/resources/http-request-query-params-config.xml]: Remove commented lines

File: extensions/http/src/main/java/org/mule/extension/http/api/listener/builder/HttpListenerResponseBuilder.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import java.util.function.Function;
+
+import javax.activation.DataHandler;
+
+/**
+ * Component that specifies how to create a proper HTTP response.
+ *
+ * @since 4.0
+ */
+@Alias("simple-response-builder")
```
Comment (pablokraan) [inline extensions/http/src/main/java/org/mule/extension/http/api/listener/builder/HttpListenerResponseBuilder.java]: Why "simple"? Is there a "complex-response-bulder"?

File: tests/http/src/test/resources/http-request-query-params-config.xml (hunk 1 above)
Comment (afelisatti) [inline tests/http/src/test/resources/http-request-query-params-config.xml]: I'm keeping the lines commented to have a clear idea of what we should test once we support query params collections.

File: extensions/http/src/main/java/org/mule/extension/http/api/listener/builder/HttpListenerResponseBuilder.java (hunk 2 above)
Comment (afelisatti) [inline extensions/http/src/main/java/org/mule/extension/http/api/listener/builder/HttpListenerResponseBuilder.java]: No, it's just to standardise both listener and request side builders. These should not survive long in any case.

**Pull request merged**